CoreErrorReportView/src/au/gov/asd/tac/constellation/views/errorreport/ErrorReportTopComponent.java

```diff
@@ -0,0 +1,894 @@
+                
+        severityItem = new CustomMenuItem(severeRepCheckBox);
+        severityItem.setHideOnClick(false);
+        filterControl.getItems().add(severityItem);
+        severityItem = new CustomMenuItem(warningRepCheckBox);
+        severityItem.setHideOnClick(false);
+        filterControl.getItems().add(severityItem);
+        severityItem = new CustomMenuItem(infoRepCheckBox);
+        severityItem.setHideOnClick(false);
+        filterControl.getItems().add(severityItem);
+        severityItem = new CustomMenuItem(fineRepCheckBox);
+        severityItem.setHideOnClick(false);
+        filterControl.getItems().add(severityItem);
+
+        ToggleGroup popupFrequency = new ToggleGroup();
+        
+        final RadioMenuItem neverItem = new RadioMenuItem("0 : Never Show Popups");
+        neverItem.setOnAction((final ActionEvent event) -> {popupMode = 0; popupControl.setText("Popup Mode : 0 ");} );
+        neverItem.setToggleGroup(popupFrequency);
+        final RadioMenuItem oneItem = new RadioMenuItem("1 : Show one popup only");
+        oneItem.setOnAction((final ActionEvent event) -> {popupMode = 1; popupControl.setText("Popup Mode : 1 ");} );
+        oneItem.setToggleGroup(popupFrequency);
+        final RadioMenuItem oneRedispItem = new RadioMenuItem("2 : Show one popup, redisplayable");
+        oneRedispItem.setOnAction((final ActionEvent event) -> {popupMode = 2; popupControl.setText("Popup Mode : 2 ");} );
+        oneRedispItem.setToggleGroup(popupFrequency);
+        final RadioMenuItem multiItem = new RadioMenuItem("3 : Show one popup per source (max 5) ");
+        multiItem.setOnAction((final ActionEvent event) -> {popupMode = 3; popupControl.setText("Popup Mode : 3 ");} );
+        multiItem.setToggleGroup(popupFrequency);
+        final RadioMenuItem multiRedispItem = new RadioMenuItem("4 : Show one per source, redisplayable");
+        multiRedispItem.setOnAction((final ActionEvent event) -> {popupMode = 4; popupControl.setText("Popup Mode : 4 ");} );
+        multiRedispItem.setToggleGroup(popupFrequency);
+        oneRedispItem.setSelected(true);
+        popupControl = new MenuButton("Popup Mode : 2 ");
+        popupControl.getItems().add(neverItem);
+        popupControl.getItems().add(oneItem);
+        popupControl.getItems().add(oneRedispItem);
+        popupControl.getItems().add(multiItem);
+        popupControl.getItems().add(multiRedispItem);
+        popupControl.setMaxWidth(200);
+        
+        final Button clearButton = new Button("Clear All Reports");
+        clearButton.setTooltip(new Tooltip("Clear all current error reports"));
+        clearButton.setOnAction((final ActionEvent event) -> {
+            // remove all matching entries in the data class
+            int errCount = sessionErrors.size();
+            for (int i = 0; i < errCount; i++) {
+                ErrorReportSessionData.getInstance().removeEntry(sessionErrors.get(i).getEntryId());
```

Comment:
would a for each make more sense to use here given this is the only use of the iterator?